src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -208,5 +227,130 @@ static void VerifyMemberOnlyApplyToTypesOrStrings (SymbolAnalysisContext context
+				var overriddenParamHasDam = overriddenTypeParam.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var overriddenTypeParamAttribute);
+				if ((methodParamHasDam && overriddenParamHasDam && !DamAttributesMatch (methodTypeParamAttribute!, overriddenTypeParamAttribute!))
+					|| (methodParamHasDam ^ overriddenParamHasDam)) {
+					context.ReportDiagnostic (Diagnostic.Create (
+						DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersMismatchOnGenericParameterBetweenOverrides),
+						methodTypeParam.Locations[0],
+						methodTypeParam.GetDisplayName (), method.GetDisplayName (), overriddenTypeParam.GetDisplayName (), overriddenMethod.GetDisplayName ()));
+				}
+			}
+
+			var basestTypeSymbol = context.Symbol.ContainingType;
+			while (basestTypeSymbol?.BaseType != null) {
+				if (basestTypeSymbol.Name == "Type" && basestTypeSymbol.ContainingNamespace.Name == "System") {
+					var methodHasDam = method.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var methodAttribute);
+					var overriddenMethodHasDam = overriddenMethod.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var overriddenMethodAttribute);
+					if ((methodHasDam && overriddenMethodHasDam && !DamAttributesMatch (methodAttribute!, overriddenMethodAttribute!))
+						|| (methodHasDam ^ overriddenMethodHasDam)) {
+						context.ReportDiagnostic (Diagnostic.Create (
+							DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersMismatchOnImplicitThisBetweenOverrides),
+							method.Locations[0],
+							method.GetDisplayName (), overriddenMethod.GetDisplayName ()));
+					}
+				}
+				basestTypeSymbol = basestTypeSymbol.BaseType;
+			}
+		}
+
+		static void VerifyDamOnInterfaceAndImplementationMethodsMatch (INamedTypeSymbol type, SymbolAnalysisContext context)
+		{
+			foreach (var (interfaceMember, implementationMember) in type.GetMemberInterfaceImplementationPairs ()) {
+				if (implementationMember is IMethodSymbol implementationMethod
+					&& interfaceMember is IMethodSymbol interfaceMethod)
+					VerifyMethodDamAttributesMatch (implementationMethod, interfaceMethod, context);
+			}
+		}
+
+		static void VerifyDamOnPropertyAndAccessorMatch (IMethodSymbol methodSymbol, SymbolAnalysisContext context)
+		{
+			if (!(methodSymbol.AssociatedSymbol?.TryGetAttribute (DynamicallyAccessedMembersAttribute, out var propertyAttribute) == true)
+				|| (propertyAttribute.ConstructorArguments.Length == 1 && propertyAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None))
+				return;
+
+			if (methodSymbol.MethodKind == MethodKind.PropertyGet
+				&& methodSymbol.TryGetReturnTypeAttribute (DynamicallyAccessedMembersAttribute, out var getValueAttribute)
+				&& !(getValueAttribute.ConstructorArguments.Length == 1 && getValueAttribute.ConstructorArguments[0].Value is (int)DynamicallyAccessedMemberTypes.None)) {
```

Comment:
Why not use the `GetDynamicallyAcessedMemberTypes` extension method which does both the "get attribute" and the extraction of the value (in this case it would be `GetDynamicallyAccessedMemberTypesOnReturnValue`).

Reply:
I didn't know that existed, thank you!